src/ms3/parse.py

```diff
@@ -565,31 +567,52 @@ def add_dir(self, directory, key=None, file_re=None, folder_re='.*', exclude_re=
-                    paths = sorted(scan_directory(d, file_re=file_re, folder_re=folder_re, exclude_re=exclude_re,
-                                                  recursive=recursive, logger=self.logger))
-                    k = os.path.basename(d)
-                    _ = self.add_files(paths=paths, key=k)
-        else:
-            self.logger.debug(f"Grouping all detected files under the key {key}.")
-            paths = sorted(scan_directory(directory, file_re=file_re, folder_re=folder_re, exclude_re=exclude_re, recursive=recursive, logger=self.logger))
-            _ = self.add_files(paths=paths, key=key)
 
+        if key is not None:
+            # contained files to be added to an existing corpus
+            paths = sorted(
+                scan_directory(directory, file_re=file_re, folder_re=folder_re,
+                               exclude_re=exclude_re,
+                               recursive=recursive, logger=self.logger))
+            _ = self.add_files(paths=paths, key=key) # makes sure that the key exists
+            return
+
+        # new corpus/corpora to be added
+        directories = sorted(iterate_corpora(directory, logger=self.logger))
+        n_corpora = len(directories)
+        if n_corpora == 0:
+            self.logger.debug(f"Treating {directory} as corpus.")
+            directories = [directory]
+        else:
+            self.logger.debug(f"{n_corpora} corpora detected.")
+            directories = [d for d in directories if re.search(exclude_re, os.path.basename(d)) is None]
+        for d in directories:
+            self.add_corpus(d, file_re=file_re, folder_re=folder_re, exclude_re=exclude_re, recursive=recursive)
+
+
+    def add_corpus(self, directory, file_re=None, folder_re='.*', exclude_re=None, recursive=True):
+        paths = sorted(
+            scan_directory(directory, file_re=file_re, folder_re=folder_re, exclude_re=exclude_re,
+                           recursive=recursive, logger=self.logger))
+        key = os.path.basename(directory)
+        self.logger.debug(f"Adding {directory} as corpus with key {key}.")
+        self.files[key] = []
+        added_ids = self.add_files(paths=paths, key=key)
+        ignored_warnings_files = self.files[key].count('IGNORED_WARNINGS')
+        if ignored_warnings_files == 0:
+            return
+        if ignored_warnings_files > 1:
+            self.logger.error(f"More than one IGNORED_WARNINGS file detected for '{key}'")
+            return
+        i = self.files[key].index('IGNORED_WARNINGS')
+        ignored_warnings = {} # parse self.full_paths[key][i] into a {logger_name -> [message_id]} dict
+        new_loggers = [self.logger_names[id] for id in added_ids]
+        for new_logger_name in new_loggers:
+            for to_be_configured, message_ids in ignored_warnings.items():
+                if to_be_configured.startswith(new_logger_name):
+                    pass # _ = get_logger(new_logger_name, ignore_warnings=message_ids)
```

Comment:
add filter to logger here

Reply:
Should we add filter to logger in parse.py not in logger.py?

Reply:
The configuration itself happens in logger.py but it needs to be triggered from this place here